policies/recipes/recipedb.go

```diff
@@ -32,16 +32,37 @@ var (
 	dbFileName   = "osconfig_recipedb"
 )
 
-// RecipeDB represents local state of installed recipes.
-type RecipeDB struct {
-	recipes map[string]Recipe
+type recipeVersion []int
+
+func (v recipeVersion) String() string {
+	if len(v) == 0 {
+		return ""
+	}
+	res := fmt.Sprintf("%d", v[0])
+	for _, val := range v[1:] {
+		res = fmt.Sprintf("%s.%d", res, val)
+	}
+	return res
 }
 
-func newRecipeDB() (*RecipeDB, error) {
+// A Recipe represents one recipe installed on the system.
+type Recipe struct {
+	Name        string
+	Version     recipeVersion
+	InstallTime int64
+	Success     bool
+}
+
+// RecipeDB represents local state of installed recipes.
+type RecipeDB map[string]Recipe
+
+// NewRecipeDB instantiates a recipeDB.
+func NewRecipeDB() (RecipeDB, error) {
```

Comment:
Is there a reason for this being public? Or really anything in this package other than the entry point, does something in guestpolicies actually need to call other functions/structs?

Reply:
Nothing needs to be public besides InstallRecipe and struct /struct attributes for marshaling.